src/base/utils/misc.cpp

```diff
@@ -620,8 +622,50 @@ QString Utils::Misc::zlibVersionString()
 }
 
 #ifdef Q_OS_WIN
+Path Utils::Misc::windowsSystemPath()
+{
+    static const Path path = []() -> Path
+    {
+        WCHAR systemPath[MAX_PATH] = {0};
+        GetSystemDirectoryW(systemPath, sizeof(systemPath) / sizeof(WCHAR));
+        return Path(QString::fromWCharArray(systemPath));
+    }();
+    return path;
+}
+#endif // Q_OS_WIN
+
+#if defined(Q_OS_MACOS) || defined(Q_OS_WIN)
 bool Utils::Misc::applyMarkOfTheWeb(const Path &file, const QString &url)
 {
+    Q_ASSERT(url.isEmpty() || url.startsWith(u"http:") || url.startsWith(u"https:"));
+
+#ifdef Q_OS_MACOS
+    // References:
+    // https://searchfox.org/mozilla-central/rev/ffdc4971dc18e1141cb2a90c2b0b776365650270/xpcom/io/CocoaFileUtils.mm#230
+    // https://github.com/transmission/transmission/blob/f62f7427edb1fd5c430e0ef6956bbaa4f03ae597/macosx/Torrent.mm#L1945-L1955
+
+    CFMutableDictionaryRef properties = ::CFDictionaryCreateMutable(kCFAllocatorDefault, 0
+        , &kCFTypeDictionaryKeyCallBacks, &kCFTypeDictionaryValueCallBacks);
+    if (properties == NULL)
```

Comment:
Wouldn't `nullptr` work?

Reply:
I don't really had time to dig in the types to be sure... so maybe let's keep the current (at least this is what the documentation guarantees: https://developer.apple.com/documentation/corefoundation/1516791-cfdictionarycreatemutable?language=objc)